### PR TITLE
tiltfile: add image deps to k8s_custom_deploy

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/kapp-imagemap2:

e7778b5c430cf2cb4c76e1407cdf683e44898fec (2021-12-16 16:52:19 -0500)
tiltfile: add image deps to k8s_custom_deploy

bd66090651852012ec96ff42f5df9871446ba176 (2021-12-16 16:52:00 -0500)
kubernetesapply: add images to custom commands

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: internal/tiltfile/value/image.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		return nil
+	}
+
+	var iter starlark.Iterator
+	switch x := v.(type) {
+	case *starlark.List:
+		iter = x.Iterate()
+	case starlark.Tuple:
+		iter = x.Iterate()
+	default:
```
Comment (milas) [inline internal/tiltfile/value/image.go:27]: IIRC if the arg is omitted, `v` will be `nil`, while if `image_deps=None` gets passed explicitly, it'll be `starlark.None`, so it'd be nice to treat that as an empty list here

File: internal/tiltfile/value/image.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package value
+
+import (
+	"fmt"
+
+	"github.com/docker/distribution/reference"
+	"go.starlark.net/starlark"
+)
+
+type ImageList []reference.Named
```
Comment (milas) [inline internal/tiltfile/value/image.go:10]: 👏

**Review by milas — APPROVED** [review]

File: internal/tiltfile/k8s_custom_deploy.go [hunk 3]
```diff
@@ -45,6 +46,7 @@ func (s *tiltfileState) k8sCustomDeploy(thread *starlark.Thread, fn *starlark.Bu
 		"delete_env?", &deleteCmdEnv,
 		"delete_cmd_bat?", &deleteCmdBatVal,
 		"container_selector?", &containerSelector,
+		"image_deps", &imageDeps,
```
[the remark this answers is not in the document]
Comment (nicks) [reply internal/tiltfile/k8s_custom_deploy.go]: yep!

File: internal/tiltfile/value/image.go (hunk 1 above)
Comment (nicks) [reply internal/tiltfile/value/image.go:29]: good catch! fixed

**Pull request merged**